src/main/g8/scalafix/build.sbt

```diff
@@ -14,41 +18,94 @@ inThisBuild(
-    scalacOptions ++= List(
-      "-Yrangepos",
-      "-P:semanticdb:synthetics:on"
-    )
+    semanticdbEnabled := true,
+    semanticdbVersion := scalafixSemanticdb.revision
   )
 )
 
-publish / skip := true
+lazy val `$repo;format="normalize"$` = (project in file("."))
+  .aggregate(
+    rules.projectRefs ++
+      input.projectRefs ++
+      output.projectRefs ++
+      tests.projectRefs: _*
+  )
+  .settings(
+    publish / skip := true
+  )
 
-lazy val rules = project.settings(
-  moduleName := "scalafix",
-  libraryDependencies += "ch.epfl.scala" %% "scalafix-core" % V.scalafixVersion
-)
+lazy val rules = projectMatrix
+  .settings(
+    moduleName := "scalafix",
+    libraryDependencies += "ch.epfl.scala" %% "scalafix-core" % V.scalafixVersion
+  )
+  .defaultAxes(VirtualAxis.jvm)
+  .jvmPlatform(rulesCrossVersions)
 
-lazy val input = project.settings(
-  publish / skip := true
-)
+lazy val input = projectMatrix
+  .settings(
+    publish / skip := true
+  )
+  .jvmPlatform(
+    scalaVersions = rulesCrossVersions,
+    settings = Seq(
+      scalacOptions += "-P:semanticdb:synthetics:on"
+    )
+  )
+  .defaultAxes(VirtualAxis.jvm)
+  .jvmPlatform(scalaVersions = rulesCrossVersions :+ scala3Version)
 
-lazy val output = project.settings(
-  publish / skip := true
-)
+lazy val output = projectMatrix
+  .settings(
+    publish / skip := true
+  )
+  .defaultAxes(VirtualAxis.jvm)
+  .jvmPlatform(scalaVersions = rulesCrossVersions :+ scala3Version)
 
-lazy val tests = project
+lazy val testsAggregate = Project("tests", file("target/testsAggregate"))
+  .aggregate(tests.projectRefs: _*)
+
+lazy val tests = projectMatrix
   .settings(
     publish / skip := true,
     libraryDependencies += "ch.epfl.scala" % "scalafix-testkit" % V.scalafixVersion % Test cross CrossVersion.full,
-    Compile / compile :=
-      (Compile / compile).dependsOn(input / Compile / compile).value,
     scalafixTestkitOutputSourceDirectories :=
-      (output / Compile / unmanagedSourceDirectories).value,
+      TestkitTargetAxis
+        .resolve(output, Compile / unmanagedSourceDirectories)
+        .value,
     scalafixTestkitInputSourceDirectories :=
-      (input / Compile / unmanagedSourceDirectories).value,
+      TestkitTargetAxis
+        .resolve(input, Compile / unmanagedSourceDirectories)
+        .value,
     scalafixTestkitInputClasspath :=
-      (input / Compile / fullClasspath).value,
+      TestkitTargetAxis.resolve(input, Compile / fullClasspath).value,
+    scalafixTestkitInputScalacOptions :=
+      TestkitTargetAxis.resolve(input, Compile / scalacOptions).value,
+    scalafixTestkitInputScalaVersion :=
+      TestkitTargetAxis.resolve(input, Compile / scalaVersion).value
+  )
+  .defaultAxes(
+    rulesCrossVersions.map(VirtualAxis.scalaABIVersion) :+ VirtualAxis.jvm: _*
+  )
+  .customRow(
+    scalaVersions = Seq(V.scala212),
+    axisValues = Seq(TestkitTargetAxis(scala3Version), VirtualAxis.jvm),
```

Comment:
I decided to have this by default in the matrix AND to keep the existing input/output skeletton in `scala` (instead of moving it to `scala-2`), so that people test against Scala 3 by default. That way, we can get more bug reports and fix whatever is missing in scalameta. Making the Scala3 run opt-in and/or having a `scala-2` folder in input/output would just delay the Scala3-readiness of community rules IMHO.